text/0000-near-shell.md

```diff
@@ -0,0 +1,449 @@
+    
+#### 2. Configuration
+
+---
+
+Configuration are key-value pairs that relate to development and deployment of smart contracts and refer to *connection* information.
+
+Example: As a user, I want to deploy a contract to my localnet instead of testnet during development. I will provide flags and/or modify configuration so that `near-shell` can determine where to connect, for what contract, and on behalf of which NEAR account.
+
+Contains:
+
+* `networkId`       : Similar to an environment to work on (ex: 'staging')
+* `nodeUrl`         : URL to the NEAR node
+* `contractName`    : NEAR account name for the smart contract
+* `walletUrl`       : URL to NEAR Wallet
+* `helperUrl`       : URL to the contract helper that provides tokens upon account creation
+* `masterAccount`   : When creating "child" accounts, `masterAccount` is the "parent" and source of initial Ⓝ balance
+
+As shown in the directory structure earlier, this file is located in the project directory at:
+
+`.near-config/connections/default.js`
+
+If a command is run and there exists no file in the path mentioned above, `near-shell` will prompt the user, suggesting it be created. 
+
+**Note**: at the time of this writing, configuration is read from the project level at:
+
+`src/config.js`
+
+This is a file containing several environment configurations with a default, which the user can choose by setting the environment variable `NODE_ENV`. Configurations will no longer be set in that manner, but stored in files.
+
+For example, the command:
+
+`near contract view near-game topPlayers --env localnet`
+
+will call the function `topPlayers` on the contract `near-game` that is deployed to localnet. The connection information will be read from the file: `.near-config/connections/localnet.js`.
+
+Users may add, remove, or modify connection environments using commands detailed later in this spec.
+
+#### 3. Key management
+[key-management]: #key-management
+
+---
+
+The storage of an account id and a corresponding private key. When a user runs `near account login` the default destination is the home directory. This can be overridden as explained in a moment.
+    
+The key file file contains the keys:
+* `type`        : Options include 
+    - `unencrypted`     : An unencrypted file containing an account's private key
+    - `native_osx`      : Private key stored with OS X key management 
+    - `native_linux`    : Private key stored with Linux-based system's key management 
+    - `native_windows`  : Private key stored with Windows key management
+* `account_id`  : The NEAR account name
+* `network`     : The NEAR network (example, "betanet", "testnet", etc.)
+* `private_key` : The plain-text private key, used when `type = "unencrypted"` 
+
+`near-shell` will look for keys in a specific order. This list is in the prioritized order and can be understood to mean, "if the key is not found here, then try the next location/store."
```

Comment:
What do you think about not having a complex order, but instead a simple default + a way to set this in config

Reply:
Do we even need complex configurable order? Seems like it always can be:

project folder -> global folder -> protected OS key store 

External stuff like `Ledger` likely makes sense to specify explicitly (as it's generally not going to be based on account ID).